### PR TITLE
Add AppImage support

### PR DESCRIPTION
**Pull request by mmatyas**

### CODE REVIEW

File: etc/appimage/travis.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+APP=openblok
+LOWERAPP=${APP,,}
+
+GIT_REV=$(git rev-parse --short HEAD)
+echo $GIT_REV
+
+make install DESTDIR=$HOME/$APP/$APP.AppDir
+
+mkdir -p $HOME/$APP/$APP.AppDir/usr/{bin,games,share}
```
Comment (probonopd) [inline etc/appimage/travis.sh]: For AppRun to pick it up, might need to move usr/games/* to usr/bin/

**Comment by mmatyas** [comment]
Thanks! This indeed fixed an issue. The AppImage created by Travis now fails with the following error:

```
/bin/bash: ./lib/x86_64-linux-gnu/libtinfo.so.5: no version information available (required by /bin/bash)
zenity, kdialog, Xdialog missing. Skipping ./bin//openblok.wrapper.
/tmp/.mount_SdwHPD/usr/bin/openblok: error while loading shared libraries: libpulsecommon-4.0.so: cannot open shared object file: No such file or directory
```

on Ubuntu 16.04, x64. I guess `libtinfo` isn't critical, and the other problem is that I've forgot to include the libs SDL requires. But since SDL can use a lot of libs ([ldd](http://pastebin.com/1F57Ehw7)), can I copy all of them into the AppImage, and rely on the blacklist to remove them, or should I find out the minimal amount of dependencies required to run the game?

**Comment by probonopd** [comment]
I'd recommend to copy in everything `ldd` has to offer, and then delete using the excludelist, you can use `functions.sh` `delete_blacklisted` like [here](https://github.com/scribusproject/scribus/blob/8a5cab8b7e0d24551d155994b61aaed7b617a3bf/bundle.sh#L67).

**Comment by mmatyas** [comment]
Hmm now I have a [weird error in wget](https://travis-ci.org/mmatyas/openblok/jobs/199694979#L2631) on Travis, even though [this](https://github.com/mmatyas/openblok/pull/9/commits/b5b872dbf63d065b39470af9a63913b77b0f65df) all I've changed. Have you ever seen something similar?

**Comment by mmatyas** [comment]
Interesting, the `generate_appimage` function failed when I tried creating the AppImage offline, but if I run the commands manually (`wget` the AppImageAssistant, etc.), it works fine, and I've managed to create an AppImage. However, when I try to start it, I get another crash:

```
zenity, kdialog, Xdialog missing. Skipping ./bin//openblok.wrapper.
[i][main] OpenBlok, created by Mátyás Mustoha, v0.6.0+bddf094
[i][init] Initializing video...
openblok: gconv_conf.c:506: __gconv_get_path: Assertion `cwd != NULL' failed.
Félbeszakítva (core készült)
```

**Comment by probonopd** [comment]
It seems like you haven't deleted the blacklisted libraries. `usr/lib/libc.so.6` is guaranteed to crash if you don't also bundle the loader and gconv (if the last sentence doesn't make sense to you, then you are better off not trying to bundle libc.so.6).

Use the `delete_blacklisted` function from functions.sh before calling `generate_appimage`.

**Comment by mmatyas** [comment]
Sorry for the late reply, I've updated the script, and now everything seems to work fine, thanks for the help!

**Comment by probonopd** [comment]
Thanks @mmatyas. Is the AppImage available for download somewhere?

**Comment by mmatyas** [comment]
I've uploaded the AppImage to the [releases page](https://github.com/mmatyas/openblok/releases); so far I've tested it on Ubuntu 16.04, 16.10, Debian Jessie, Fedora 25 Workstation and Manjaro 16.10. It can also run on older distros (tested 14.04), but requires at least GLIBCXX_3.4.20 (GCC 4.9).

**Pull request merged**